nautobot/dcim/models/device_component_templates.py

```diff
@@ -249,11 +250,35 @@ class Meta:
         unique_together = ("device_type", "name")
 
     def instantiate(self, device):
+        record = {}
+        content_type = ContentType.objects.get_for_model(Interface)
+        fields = CustomField.objects.filter(content_types=content_type)
+        for field in fields:
+            record[field.name] = field.default
+
```

Comment:
Hi Glenn @glennmatthews, thank you for the advice. Did I understand you correctly?

Reply:
I thought your previous approach of loading the custom field data during `instantiate_model` made sense, I'm not sure I see the benefit of doing this in `instantiate` as you are now.

My point was just that the `_custom_field_data` that you pass through to the `model()` call (e.g., the `_custom_field_data` for an `Interface` instance being created) is *not* the same as `self._custom_field_data` (e.g. the `_custom_field_data` on the `InterfaceTemplate` that's creating the `Interface`). In other words, `instantiate()`/`instantiate_model()` on the `ComponentTemplateModel` should *not* modify `self._custom_field_data`, nor should `self._custom_field_data` be passed to the `model()` call.